src/providers/wfs/qgswfsshareddata.cpp

```diff
@@ -466,6 +477,26 @@ bool QgsWFSSharedData::createCache()
     return false;
   }
 
+  // The id_cache should be generated once for the lifetime of QgsWFSConstants
+  // to ensure consistency of the ids returned to the user.
+  if ( mCacheIdDbname.isEmpty() )
+  {
+    mCacheIdDbname = QDir( cacheDirectory ).filePath( QStringLiteral( "wfs_id_cache_%1.sqlite" ).arg( tmpCounter ) );
+    Q_ASSERT( !QFile::exists( mCacheIdDbname ) );
+    if ( mCacheIdDb.open( mCacheIdDbname ) != SQLITE_OK )
+    {
+      QgsMessageLog::logMessage( tr( "Cannot create temporary id cache" ), tr( "WFS" ) );
+      return false;
+    }
+    ( void )sqlite3_exec( mCacheIdDb.get(), "PRAGMA synchronous=OFF", nullptr, nullptr, nullptr );
```

Comment:
Does the `mCacheIdDb.exec( QStringLiteral( "PRAGMA synchronous=off" ) )` work?

Reply:
There's no such method in sqlite3_database_unique_ptr. If i added one, that would complicate backportability unless I also backport it

Reply:
https://github.com/qgis/QGIS/blob/master/src/core/qgssqliteutils.h#L150

Reply:
Oh, I actually looked at qgsqliteutils.h from 3.4. Would backporting this new method be acceptable ? This doesn't affect backward compatibility.

Reply:
Sounds good to me, yes!